### ISSUE TITLE
schema: release-23.1 add primary key DML injection failure

### ISSUE THREAD

**Issue by ecwall** [cockroachdb/cockroach#111832]
Note: This only affects release-23.1 and works properly on master.

Setup
```
CREATE TABLE tbl (
	insert_phase_ordinal TEXT NOT NULL, 
	operation_phase_ordinal TEXT NOT NULL,
	operation TEXT NOT NULL,
	val INT
);
```

Schema change
```
ALTER TABLE tbl ADD PRIMARY KEY (insert_phase_ordinal, operation_phase_ordinal, operation);
```

Schema change DML injection failure
```
W231005 13:33:07.622796 2250 sql/schemachanger/scrun/scrun.go:139 ⋮ [T1,n1,job=‹NEW SCHEMA CHANGE id=905846408666349569›] 205  failed executing declarative schema change PostCommitPhase stage 15 of 15 with 1 ValidationType op (rollback=false) for ALTER TABLE with error: duplicate key value violates unique constraint ‹"tbl_pkey"›
```

Jira issue: CRDB-32086

**Comment by rafiss** [comment]
@ecwall does this happen when performing any INSERT? also could you link to the test and the command to run it, so it's easier to reproduce this?

**Comment by ecwall** [comment]
I am not sure if any insert causes it to fail, but this test case consistently fails on `release-23.1` (I have just been running it in GoLand so I can debug it easily): https://github.com/cockroachdb/cockroach/blob/5811f3fec5dfb209fd06f56344113723f6354e19/pkg/sql/schemachanger/dml_injection_test.go#L267

**Comment by rafiss** [comment]
This is fixed in 23.2 and later. We won't prioritize fixing 23.1 unless we hear feedback on this.
